js/src/elements/tab-container/element.ts

```diff
@@ -0,0 +1,471 @@
+		if (typeof listener === 'function') {
+			this.addEventListener(
+				'tab-container-change',
+				listener as unknown as EventListenerOrEventListenerObject
+			);
+		}
+	}
+
+	#onTabContainerChanged: ((event: TabContainerChangeEvent) => void) | null = null;
+	get onTabContainerChanged() {
+		return this.#onTabContainerChanged;
+	}
+
+	set onTabContainerChanged(listener: ((event: TabContainerChangeEvent) => void) | null) {
+		if (this.#onTabContainerChanged) {
+			this.removeEventListener(
+				'tab-container-changed',
+				this.#onTabContainerChanged as unknown as EventListenerOrEventListenerObject
+			);
+		}
+		this.#onTabContainerChanged =
+			typeof listener === 'object' || typeof listener === 'function' ? listener : null;
+		if (typeof listener === 'function') {
+			this.addEventListener(
+				'tab-container-changed',
+				listener as unknown as EventListenerOrEventListenerObject
+			);
+		}
+	}
+
+	get onChanged() {
+		return this.onTabContainerChanged;
+	}
+
+	set onChanged(listener: ((event: TabContainerChangeEvent) => void) | null) {
+		this.onTabContainerChanged = listener;
+	}
+
+	// eslint-disable-next-line @typescript-eslint/member-ordering
+	static observedAttributes = ['vertical'];
+
+	get #tabList(): HTMLElement {
+		const wrapper = this.querySelector('[slot=tablist-wrapper]');
+		if (wrapper?.closest(this.tagName) === this) {
+			return wrapper.querySelector('[role=tablist]') as HTMLElement;
+		}
+		const slot = this.#tabListSlot;
+		if (this.#tabListTabWrapper.hasAttribute('role')) {
+			return this.#tabListTabWrapper;
+		} else {
+			return slot.assignedNodes()[0] as HTMLElement;
+		}
+	}
+
+	get #tabListWrapper() {
+		return this.shadowRoot!.querySelector<HTMLSlotElement>('slot[part="tablist-wrapper"]')!;
+	}
+
+	get #tabListTabWrapper() {
+		return this.shadowRoot!.querySelector<HTMLElement>('div[part="tablist-tab-wrapper"]')!;
+	}
+
+	get #beforeTabsSlot() {
+		return this.shadowRoot!.querySelector<HTMLSlotElement>('slot[part="before-tabs"]')!;
+	}
+
+	get #afterTabsSlot() {
+		return this.shadowRoot!.querySelector<HTMLSlotElement>('slot[part="after-tabs"]')!;
+	}
+
+	get #afterPanelsSlot() {
+		return this.shadowRoot!.querySelector<HTMLSlotElement>('slot[part="after-panels"]')!;
+	}
+
+	get #tabListSlot() {
+		return this.shadowRoot!.querySelector<HTMLSlotElement>('slot[part="tablist"]')!;
+	}
+
+	get #panelSlot() {
+		return this.shadowRoot!.querySelector<HTMLSlotElement>('slot[part="panel"]')!;
+	}
+
+	get #tabs(): HTMLElement[] {
+		if (this.#tabListTabWrapper.matches('[role=tablist]')) {
+			return this.#tabListSlot.assignedNodes() as HTMLElement[];
+		}
+		return Array.from<HTMLElement>(this.#tabList?.querySelectorAll('[role="tab"]') || []).filter(
+			tab => tab instanceof HTMLElement && tab.closest(this.tagName) === this
+		);
+	}
+
+	get #next(): HTMLButtonElement {
+		return this.querySelector<HTMLButtonElement>('[data-segmented-control-nav="next"]')!;
```

Comment:
more names that need updating `data-segmented-control` should be `data-tab-container-nav="next"` and `data-tab-container-nav="previous"`?

Reply:
I like it.  Keeps with the convention we talked about.  Making adjustment now

Reply:
✅